platform.md

```diff
@@ -446,6 +454,122 @@ Usage:
+| `<plan>`      | `CNB_PLAN_PATH`      | `<layers>/plan.toml`     | Path to resolved build plan (see [`plan.toml`](#plantoml-toml))                                                 |
+| `<log-level>` | `CNB_LOG_LEVEL`      | `info`                   | Log Level                                                                                                       |
+| `<platform>`  | `CNB_PLATFORM_DIR`   | `/platform`              | Path to platform directory                                                                                      |
+
+##### Outputs
+| Output                                       | Description                                                                   |
+|----------------------------------------------|-------------------------------------------------------------------------------|
+| [exit status]                                | (see Exit Code table below for values)                                        |
+| `/dev/stdout`                                | Logs (info)                                                                   |
+| `/dev/stderr`                                | Logs (warnings, errors)                                                       |
+| `<output>/<image extension ID>/<Dockerfile>` | Generated Dockerfiles (see [Image Extension Specfication](image-extension.md) |
+| `<layers>/config/metadata.toml`              | Build metadata (see [`metadata.toml`](#metadatatoml-toml))                    |
+
+| Exit Code       | Result                              |
+|-----------------|-------------------------------------|
+| `0`             | Success                             |
+| `11`            | Platform API incompatibility error  |
+| `12`            | Buildpack API incompatibility error |
+| `1-10`, `13-19` | Generic lifecycle errors            |
+| `91`            | Image extension build error         |
+| `90`, `92-99`   | Build-Ext-specific lifecycle errors |
+
+- The lifecycle SHALL execute all image extensions in the order defined in `<group>` according to the process outlined in the [Buildpack Interface Specification](buildpack.md).
+- The lifecycle SHALL add all invoked image extensions to`<layers>/config/metadata.toml`.
+- The lifecycle SHALL copy all output Dockerfiles to `<output>`.
+- The lifecycle SHALL record all Dockerfiles and build args in `<extender>`.
+- The lifecycle SHALL filter the build plan with dependencies provided by image extensions.
+
+#### `extender`
+
+The platform MAY execute `extender` for Linux builds
+
+Usage:
+
+```
+/cnb/lifecycle/extender \
+  [-analyzed <analyzed>] \
+  [-app <app>] \
+  [-cache-image <cache-image>] \
+  [-extender <extender> ] \
+  [-genpkgs <genpkgs> ] \
+  [-ignore-paths <ignore-paths>] \
+  [-kind <kind>] \
+  [-log-level <log-level>] \
+  [-work-dir <work-dir>] \
+  <base-image> \
+  <target-image>
```

Comment:
This interface is a bit weird... I can't remember why it ended up this way. Perhaps better would be to accept either a `-build-image` or `-run-image` flag (from which the `kind` can be inferred) and default to `kind=run` when no flag is provided, reading the run image from analyzed.toml.